demo/screens/EthereumSignUp.js

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import {
+  Image,
+  Button,
+  StyleSheet,
+  View,
+  Text,
+  TextInput,
+  KeyboardAvoidingView,
+} from "react-native";
+import {
+  createStackNavigator,
+  createAppContainer,
+  Header,
+} from "react-navigation";
+
+export default class EthereumSignUp extends React.Component {
+  state = {
+    text: "",
+  };
+
+  render() {
+    return (
+      <KeyboardAvoidingView
+        keyboardVerticalOffset={Header.HEIGHT + 20}
+        style={styles.container}
+        behavior="padding"
+      >
+        <Text style={styles.text}>
+          {`Cool. Let's start by picking your Tasit username.`}
+        </Text>
+        <View style={styles.userRow}>
+          <View style={styles.userInputView}>
+            <TextInput
+              style={styles.userInput}
+              onChangeText={text => this.setState({ text })}
+              value={this.state.text}
+              placeholder="username"
+            />
+          </View>
+          <View style={styles.ensView}>
+            <Text style={styles.ensText}>{`.tasitid.eth`}</Text>
+          </View>
+        </View>
+
+        <View style={styles.buttonView}>
+          <Button title="Continue" onPress={() => {}} />
+        </View>
+      </KeyboardAvoidingView>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: "#fff",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+  text: {
+    padding: 10,
+    fontSize: 30,
+    textAlign: "center",
+    fontWeight: "bold",
+    color: "gray",
+  },
+  userRow: { flexDirection: "row" },
+  userInputView: { flex: 1, alignItems: "flex-end" },
+  userInput: { justifyContent: "flex-start", width: 90, fontSize: 20 },
```

Comment:
When there are this many styles, I think it's an indication that this component should be sub-component-ized

Reply:
Done

Reply:
I've extract inputtext + button as a form components. Tell me if you think that is better break them more.

Reply:
Okay, I'll take a look, but that sounds like that should be good! Thanks.